RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc

```diff
@@ -637,6 +636,10 @@ void MuonIdProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
      }
    }
 
+   // muon arbitration
+   fillArbitrationInfo( outputMuons.get() );
+   arbitrateMuons( outputMuons.get(), caloMuons.get() );
```

Comment:
I looked at implementing it - it's too much hassle. The module is used in tons of places (HLT) and cloned in even larger number of places. Providing it as an untracked parameter with a default is wrong, because it's important parameter. So I think it should stay as is, i.e. TrackerMuons will get arbitrated. I don't see why earlyMuons should not be arbitrated if they are TrackerMuons. All the zoo of muons that we have now won't be affected unless they are TrackerMuons, which is a specific algorithm and it can change.